distributed/tests/test_client.py

```diff
@@ -1550,6 +1551,38 @@ def g():
                 break
 
 
+@gen_cluster(client=True)
+def test_upload_file_egg(c, s, a, b):
+    def g():
+        import package_1, package_2
+        return package_1.a, package_2.b
+
+    for value in [123, 456]:
+        dir_contents = {
+                'setup.py': (
+                    'from setuptools import setup, find_packages\n'
+                    'setup(name="my_package", packages=find_packages())\n'
+                ),
+                'package_1/__init__.py': 'a = {}'.format(value),
+                'package_2/__init__.py': 'b = {}'.format(value),
+        }
+        with tmp_dir(dir_contents) as directory:
+            subprocess.check_call(['python', 'setup.py', 'bdist_egg'], cwd=directory)
```

Comment:
Perhaps `sys.executable` here rather than `'python'`?

Reply:
What happens if the user doesn't have install privileges to where their Python is installed, such as if they're using their system Python?

Reply:
This does not _install_ `pakage_1/package_2`. It merely _compiles_ them into an egg. You can verify this by running a simple `setup.py` on a non-root account.

```
    $ cat setup.py
    from setuptools import setup, find_packages
    setup(
        name='my_package',
        packages=find_packages(),
    )
    $ python3 setup.py bdist_egg

```